django/db/models/sql/datastructures.py

```diff
@@ -79,15 +83,35 @@ def as_sql(self, compiler, connection):
         qn = compiler.quote_name_unless_alias
         qn2 = connection.ops.quote_name
 
+        if hasattr(self.join_field, "get_joining_expressions"):
+            self.join_cols = self.join_field.get_joining_expressions(
+                connection,
+                self.parent_alias,
+                self.table_alias,
+            )
+        else:
+            warnings.warn(
+                "The get_joining_columns method is deprecated. Implement "
+                "get_joining_expressions instead.",
+                RemovedInDjango60Warning,
+            )
```

Comment:
I may have have added too many warning raises 🤔 What do you think?

(and I did not write tests for them, but I guess I should? 🙈 )